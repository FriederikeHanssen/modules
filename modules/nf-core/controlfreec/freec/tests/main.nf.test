nextflow_process {

    name "Test Process CONTROLFREEC_FREEC"
    script "../main.nf"
    process "CONTROLFREEC_FREEC"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "controlfreec"
    tag "controlfreec/freec"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("controlfreec - tumor/normal") {

        setup {
            run("UNTAR"){
                script "../../../untar/main.nf"
                process {
                    """
                    input[0] = [ [id:'archive'],
                                file(params.test_data['homo_sapiens']['genome']['genome_21_chromosomes_dir'], checkIfExists: true)
                            ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false, sex:'XX' ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_mpileup'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test2_mpileup'], checkIfExists: true),
                    [],[],[],[]
                ]

                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                input[3] = []
                input[4] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz'], checkIfExists: true)
                input[5] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz_tbi'], checkIfExists: true)
                input[6] = UNTAR.out.untar.map{ it[1] }
                input[7] = []
                input[8] = file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
                input[9] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.config.get(1)).readLines().contains("[general]") },
                { assert snapshot(
                    process.out.bedgraph,
                    process.out.control_cpn,
                    process.out.sample_cpn,
                    process.out.gcprofile_cpn,
                    process.out.BAF,
                    process.out.CNV,
                    process.out.info,
                    process.out.ratio,
                    process.out.versions).match() }
            )
        }

    }

    test("controlfreec - single") {

        setup {
            run("UNTAR"){
                script "../../../untar/main.nf"
                process {
                    """
                    input[0] = [ [id:'archive'],
                                file(params.test_data['homo_sapiens']['genome']['genome_21_chromosomes_dir'], checkIfExists: true)
                            ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test2', single_end:false, sex:'XX' ], // meta map
                    [],
                    file(params.test_data['homo_sapiens']['illumina']['test2_mpileup'], checkIfExists: true),
                    [],[],[],[]
                ]

                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                input[3] = []
                input[4] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz'], checkIfExists: true)
                input[5] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz_tbi'], checkIfExists: true)
                input[6] = UNTAR.out.untar.map{ it[1] }
                input[7] = []
                input[8] = file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
                input[9] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.config.get(1)).readLines().contains("[general]") },
                { assert snapshot(
                    process.out.bedgraph,
                    process.out.control_cpn,
                    process.out.sample_cpn,
                    process.out.gcprofile_cpn,
                    process.out.BAF,
                    process.out.CNV,
                    process.out.info,
                    process.out.ratio,
                    process.out.versions).match() }
            )
        }

    }

}
