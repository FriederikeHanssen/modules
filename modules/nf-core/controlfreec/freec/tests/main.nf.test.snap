{
    "controlfreec - tumor/normal": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_ratio.BedGraph:md5,e1d777f6c204e405a880a0b6cc45b2c8"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test.mpileup.gz_control.cpn:md5,1768b571677c418560e5a8fe203bdc79"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_sample.cpn:md5,c80dad58a77b1d7ba6d273999f4b4b4b"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_BAF.txt:md5,97614489f64ce8e7f5d4c69e0811a872"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_CNVs:md5,7fa21a187466579fa532d6d40d7412cf"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_info.txt:md5,1a3055d35028525ccc9e693cc9f335e0"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_ratio.txt:md5,0565ec12db85fb2a84b2072563f27bba"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "config.txt:md5,08c3a3e664bf99deb7fcac2ff1c2ea66"
                    ]
                ],
                "9": [
                    "versions.yml:md5,e704fc0e6d1ac333dc419498fa128769"
                ],
                "BAF": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_BAF.txt:md5,97614489f64ce8e7f5d4c69e0811a872"
                    ]
                ],
                "CNV": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_CNVs:md5,7fa21a187466579fa532d6d40d7412cf"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_ratio.BedGraph:md5,e1d777f6c204e405a880a0b6cc45b2c8"
                    ]
                ],
                "config": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "config.txt:md5,08c3a3e664bf99deb7fcac2ff1c2ea66"
                    ]
                ],
                "control_cpn": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test.mpileup.gz_control.cpn:md5,1768b571677c418560e5a8fe203bdc79"
                    ]
                ],
                "gcprofile_cpn": [
                    
                ],
                "info": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_info.txt:md5,1a3055d35028525ccc9e693cc9f335e0"
                    ]
                ],
                "ratio": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_ratio.txt:md5,0565ec12db85fb2a84b2072563f27bba"
                    ]
                ],
                "sample_cpn": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sex": "XX"
                        },
                        "test2.mpileup.gz_sample.cpn:md5,c80dad58a77b1d7ba6d273999f4b4b4b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e704fc0e6d1ac333dc419498fa128769"
                ]
            }
        ],
        "timestamp": "2023-12-05T20:07:42.461666506"
    }
}